### PR TITLE
fix projects when watching with selector

### PR DESCRIPTION
**Pull request by juanvallejo**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1633101

Fixes fieldSelector matching when watching Projects.

cc @deads2k @soltysh

### CODE REVIEW

**Comment by enj** [comment]
@openshift/sig-security

File: pkg/project/auth/watch.go [hunk 1]
```diff
@@ -131,7 +139,7 @@ func (w *userProjectWatcher) GroupMembershipChanged(namespaceName string, users,
 			w.cacheError <- errors.New("delete notification timeout")
 		}
 
-	case hasAccess:
+	case hasAccess && known:
 		namespace, err := w.projectCache.GetNamespace(namespaceName)
```
Comment (enj) [inline pkg/project/auth/watch.go]: Spoke with @juanvallejo, he should be moving the predicate logic here

File: pkg/project/apiserver/registry/project/strategy.go [hunk 2]
```diff
@@ -77,3 +82,30 @@ func (projectStrategy) Canonicalize(obj runtime.Object) {
 func (projectStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
 	return validation.ValidateProjectUpdate(obj.(*projectapi.Project), old.(*projectapi.Project))
 }
+
+// GetAttrs returns labels and fields of a given object for filtering purposes.
+func getAttrs(obj runtime.Object) (labels.Set, fields.Set, bool, error) {
```
[the remark this answers is not in the document]
Comment (juanvallejo) [reply pkg/project/apiserver/registry/project/strategy.go]: No sense in doing so, I think. They're only used within this file.
Can make public in future if a consumer outside of this package ever needs them

**Comment by juanvallejo** [comment]
Looks like extended_clusterup failure might be unrelated to these changes

File: pkg/project/util/util.go [hunk 3]
```diff
@@ -45,3 +53,30 @@ func ConvertNamespaceList(namespaceList *kapi.NamespaceList) *projectapi.Project
 	}
 	return projects
 }
+
+// GetAttrs returns labels and fields of a given object for filtering purposes.
+func getAttrs(obj runtime.Object) (labels.Set, fields.Set, bool, error) {
```
Comment (soltysh) [inline pkg/project/util/util.go]: Nit: uppercase vs lowercase function name vs godoc.

File: pkg/project/util/util.go [hunk 4]
```diff
@@ -45,3 +53,30 @@ func ConvertNamespaceList(namespaceList *kapi.NamespaceList) *projectapi.Project
...
+func MatchProject(label labels.Selector, field fields.Selector) apistorage.SelectionPredicate {
+	return apistorage.SelectionPredicate{
+		Label:    label,
+		Field:    field,
+		GetAttrs: getAttrs,
+	}
+}
+
+// NamespaceToSelectableFields returns a field set that represents the object
+func projectToSelectableFields(projectObj *projectapi.Project) fields.Set {
```
Comment (soltysh) [inline pkg/project/util/util.go]: Nit: godoc does not matches function name.

**Comment by juanvallejo** [comment]
@soltysh thanks, comments addressed. Reapplying lgtm label based on previous approval

**Pull request merged**